### PR TITLE
Rename TriggerPublisherFunc as IndexerFunc

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR addresses the TODO @wojtek-t left from PR #79793

Renames TriggerPublisherFunc as IndexerFunc

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go [hunk 1]
```diff
@@ -211,7 +211,7 @@ type filterWithAttrsFunc func(key string, l labels.Set, f fields.Set) bool
 
 type indexedTriggerFunc struct {
 	indexName   string
-	triggerFunc storage.TriggerPublisherFunc
+	triggerFunc storage.IndexerFunc
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go]: If we're renaming, we should also rename the field here at least.

**Comment by wojtek-t** [comment]
/assign @jpbetz 
who request this renaming

**Comment by tedyu** [comment]
@jpbetz
See if there is anything else to be changed.

**Review by jpbetz — COMMENTED** [review]
/lgtm

**Comment by tedyu** [comment]
@wojtek-t 
Can you approve this ?

**Pull request merged**